### PR TITLE
Fix error when value come from Expectation and Symfony DomCrawler com…

### PR DESCRIPTION
**Pull request by mattheo-geoffray**
…ponent   
Related issue: https://github.com/blackfireio/player/issues/42

### CODE REVIEW

**Comment by romainneutron** [comment]
Hello, thanks for this report and patch! Let's wait for @iamluc feedback. He'll be back tomorrow

**Comment by mattheo-geoffray** [comment]
Hi @romainneutron 
No problem! Maybe there is a more generic way to test the value here to cover more possibility, let's see his feedback.   
Thanks!

**Comment by iamluc** [comment]
Hi @mattheo-geoffray 

In my opinion, instead of testing the type of the object in the `dumpEscaped()` method, we should add a new case `case is_object($value):` in method `formatResult()`

Then we could test if the object has a method `__toString`, or just use `get_class()` if that's not the case

**Comment by mattheo-geoffray** [comment]
Hi @iamluc   
Thanks for the return I will change that quickly!

**Comment by mattheo-geoffray** [comment]
Hi @iamluc !   
Do you want me to add an error in the case the object does not have a `__toString` function?    
In my opinion the user does not have to worry about the class processing in the background, so maybe a generic error like `An error occurred while formatting the result.`?

**Comment by mattheo-geoffray** [comment]
I have pushed a fix let me know if is that OK for you guys 🙂

File: Player/ExpressionLanguage/ExtractResultsVisitor.php [hunk 1]
```diff
@@ -89,11 +89,30 @@ private function formatResult($value)
...
+    {
+        if (method_exists($value, '__toString')) {
+            return $value->__toString();
+        }
+
+        if ($value instanceof \Symfony\Component\DomCrawler\Crawler) {
+            return $value->html();
+        }
+
+        return '';
```
Comment (iamluc) [inline Player/ExpressionLanguage/ExtractResultsVisitor.php]: Maybe something like 
```php
return '(object) '.get_class($value);
```

Comment (mattheo-geoffray) [reply Player/ExpressionLanguage/ExtractResultsVisitor.php]: I have pushed the fix

Comment (mattheo-geoffray) [reply Player/ExpressionLanguage/ExtractResultsVisitor.php]: I have pushed the fix

**Comment by romainneutron** [comment]
Thanks !
Merged as of https://github.com/blackfireio/player/commit/a7a0314dcf177a48af79b294c916f277604e1441

**Pull request closed without merging**